Dataviews: Fix alignment issue of "Title" column header

im3dabasia opened the pull request:
Closes: https://github.com/WordPress/gutenberg/issues/68838

## What?
This PR addresses the issue where the "Title" column header is not properly aligned in `Dataviews` when a titleField is set. The column header shifts to the left, misaligned with the search box above and the rows below.

## Why?
The issue is caused by an extra margin-left on the `dataviews-view-table-header`.

## How?
Removes the conflicting margin-left to restore proper header alignment.

## Testing Instructions

1. Run npm run storybook.
2. Open the Dataviews story in Storybook.
3. Check the header alignment for the first column.
4. Ensure the "Title" column header is now properly aligned.

## Screenshots 

|Before|After|
|-|-|
|<img width="547" alt="Screenshot 2025-01-22 at 11 25 35 PM" src="https://github.com/user-attachments/assets/81d6a978-732e-4440-a945-941d610f22aa" />|<img width="544" alt="Screenshot 2025-01-22 at 11 24 12 PM" src="https://github.com/user-attachments/assets/30ed5f59-515c-4bf7-a0b9-96c89fd2d6d2" />|

im3dabasia committed:
fix: Alignment issue of first header in dataviews

t-hamano commented:
Thanks for the PR!

My intuition is that the root problem is nested elements:

```html
<tr class="dataviews-view-table__row">
  <th scope="col">
    <span class="dataviews-view-table-header">
      <button class="dataviews-view-table-header-button">Title</button>
    </span>
  </th>
</tr>
```

Clickable headers should be rendered like this:

```html
<tr class="dataviews-view-table__row">
  <th scope="col">
    <button class="dataviews-view-table-header-button">Title</button>
  </th>
</tr>
```

It would be a good idea to investigate why you have nested elements.

hbhalodia commented:
I guess we should remove the `span` which is added as a parent of the clickable field. This is because, if we scroll a bit down, we have further columns' header which are added without any extra parent span.

On further debugging, found that this commit - https://github.com/WordPress/gutenberg/commit/6b16c7359862d7131df1c06e598b2480833f248f#diff-3021810c0ffdc124de3220e52b952b9aa2f6e604e01383e0e4c8b0aecd5e48cf, have added the `titleField` column header.

I guess if we remove this span, it would not have any regression issues and our main issue would also get resolved on `titleField` not being aligned.

Thank You,

im3dabasia committed:
fix: Remove span parent of title field

im3dabasia committed:
fix: Remove inline css and remove extra css code

im3dabasia commented:
Hey @t-hamano ,

I've updated the PR to incorporate the suggested changes. When you have a moment, please review the latest updates.

<details>
<summary> Screenshot</summary>

<img width="1007" alt="image" src="https://github.com/user-attachments/assets/9e0bb7db-7750-4ed6-baab-6675e1293661" />
</details


Additionally, I wanted to get your thoughts on two points:

1. I believe we can omit this CSS class since it was specifically targeting the <span> element we removed.

https://github.com/WordPress/gutenberg/blob/52a3c28547e5e6848702efc1f03d34d585bd0118/packages/dataviews/src/dataviews-layouts/table/style.scss#L40-L43

After searching the dataviews package, I found that the `dataviews-view-table-header` class is used for both the title and actions table headers. However, this particular CSS rule is too specific and was only affecting the deleted <span>, so it seems reasonable to remove it. For the actions header, there is already a corresponding CSS rule applied, attached down below.

https://github.com/WordPress/gutenberg/blob/52a3c28547e5e6848702efc1f03d34d585bd0118/packages/dataviews/src/dataviews-layouts/table/style.scss#L159-L161



2. Another thing I wanted to point out is that this inline style could be moved to the SCSS file. This would eliminate the warning, though I understand it’s outside the scope of this PR.

https://github.com/WordPress/gutenberg/blob/52a3c28547e5e6848702efc1f03d34d585bd0118/packages/dataviews/src/dataviews-layouts/table/index.tsx#L143-L148

Let me know what you think! Thanks for your time.

t-hamano reviewed (approved):
LGTM! I verified that no visual changes occurred other than to the title field.

> I believe we can omit this CSS class since it was specifically targeting the element we removed.

> Another thing I wanted to point out is that this inline style could be moved to the SCSS file. This would eliminate the warning, though I understand it’s outside the scope of this PR.

I think both of these are correct 👍
